### PR TITLE
Data pipeline - Scheduling Jobs

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**

- [x] Functionality to schedule jobs to stream historical data.
- [x] Adds data type validation to data fetched from device registry.
- [x] Adds error handling to the KCCA DAGs

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-1040](https://airqoteam.atlassian.net/browse/PLAT-1040)

**_HOW DO I TEST OUT THIS PR?_**

- Setup your environment. Follow instructions in the [Environment setup section of the Readme.md file](https://github.com/airqo-platform/AirQo-api/tree/fix-data-pipeline/src/airflow#environment-setup)
- Follow instructions in the [Schedule section of the Readme.md file](https://github.com/airqo-platform/AirQo-api/tree/fix-data-pipeline/src/airflow#using-the-api)

### CODE REVIEW

File: src/airflow/README.md [hunk 1]
```diff
@@ -117,9 +117,30 @@ Ctrl + c
 sh clean.sh  
 ```
 
-## Working with dags
+## Working with DAGs
 
-ETL dags for historical data require you to specify `startDateTime` and `endDateTime` in the dag config using the format `YYYY-MM-ddTHH:mm:ssZ`
+### Scheduling DAGs for historical data
```
Comment (NoahNsimbe) [inline src/airflow/README.md:122]: Documentation for scheduling jobs that stream historical data

File: src/airflow/airqo_etl_utils/arg_parse_validator.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+def valid_datetime_format(arg_str: str) -> str:
+    try:
+        datetime.strptime(arg_str, "%Y-%m-%dT%H:%M:%SZ")
+        return arg_str
+    except ValueError:
+        msg = "Invalid Datetime {0} format. Expected format: 'yyyy-MM-ddThh:mm:ssZ'".format(
+            arg_str
+        )
+        raise argparse.ArgumentTypeError(msg)
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/arg_parse_validator.py:13]: validating user input

File: src/airflow/airqo_etl_utils/data_warehouse_utils.py [hunk 3]
```diff
@@ -121,6 +120,44 @@ def extract_sites_meta_data(tenant=None) -> list:
...
+            "distance_to_nearest_tertiary_road",
+            "distance_to_nearest_primary_road",
+            "distance_to_nearest_road",
+            "distance_to_nearest_residential_road",
+            "distance_to_nearest_secondary_road",
+            "distance_to_nearest_unclassified_road",
+        ]
+    ].apply(
+        pd.to_numeric, errors="coerce"
+    )
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/data_warehouse_utils.py:159]: formatting some site information to a float data type.

File: src/airflow/airqo_etl_utils/kcca_utils.py [hunk 4]
```diff
@@ -31,11 +32,16 @@ def query_kcca_measurements(frequency: str, start_time: str, end_time: str):
...
-        print(f"{results.content}")
+    try:
+        results = requests.get(api_url, headers=headers)
+        if results.status_code != 200:
+            print(f"{results.content}")
+            return []
+        return results.json()
+    except Exception as ex:
+        traceback.print_exc()
+        print(ex)
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/kcca_utils.py:43]: Error handling on KCCA DAGs

File: src/airflow/schedule-dag.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+from airqo_etl_utils.arg_parse_validator import valid_datetime_format
+from airqo_etl_utils.date import date_to_str
+
+BASE_DIR = Path(__file__).resolve().parent
+dotenv_path = os.path.join(BASE_DIR, ".env")
+load_dotenv(dotenv_path)
+
+
+class ScheduleDag:
+    def __init__(
```
Comment (NoahNsimbe) [inline src/airflow/schedule-dag.py:22]: class with functionality for scheduling jobs for historical data

**Review by 123MwanjeMike — APPROVED** [review]

**Review by Baalmart — COMMENTED** [review]
LGTM

**Pull request merged**